qualtran/bloqs/qubitization/qubitization_walk_operator.py

```diff
@@ -153,6 +161,15 @@ def _circuit_diagram_info_(self, args: cirq.CircuitDiagramInfoArgs) -> cirq.Circ
     def adjoint(self) -> 'QubitizationWalkOperator':
         return attrs.evolve(self, uncompute=not self.uncompute)
 
+    @cached_property
+    def prepare(self) -> Union[PrepareOracle, BlackBoxPrepare]:
+        """Get the Prepare bloq if appropriate from the block encoding."""
```

Comment:
could you add a TODO that says this should use `self.signal_state`?

Reply:
Done.

Reply:
Eh, need to make the walk operator a block encoding, some issues with controls I'll leave for follow up.

Reply:
https://github.com/quantumlib/Qualtran/issues/1266